### DISCUSSION TITLE
ESP32-2432S022C Compile Errors

### DISCUSSION THREAD

**Discussion by GerdS1960** [rzeldent/esp32-smartdisplay#157] [Q&A]
After Installation of VisualCode following your tutorial i ran into problems compiling your code.

These are the errorcodes from the compiler:
[{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/include/esp_lcd_touch.h",
	"owner": "cpp",
	"severity": 8,
	"message": "'CONFIG_ESP_LCD_TOUCH_MAX_POINTS' undeclared here (not in a function); did you mean 'CONFIG_FMB_TCP_PORT_MAX_CONN'?",
	"startLineNumber": 76,
	"startColumn": 14,
	"endLineNumber": 76,
	"endColumn": 14
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_SWAP_XY\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 152,
	"startColumn": 7,
	"endLineNumber": 152,
	"endColumn": 22
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "cpp",
	"severity": 8,
	"message": "'DISPLAY_SWAP_XY' undeclared (first use in this function); did you mean 'LCD_SWAP_XY'?",
	"startLineNumber": 152,
	"startColumn": 59,
	"endLineNumber": 152,
	"endColumn": 59
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_MIRROR_X\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 153,
	"startColumn": 7,
	"endLineNumber": 153,
	"endColumn": 22
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "cpp",
	"severity": 8,
	"message": "'DISPLAY_MIRROR_X' undeclared (first use in this function); did you mean 'LCD_MIRROR_X'?",
	"startLineNumber": 153,
	"startColumn": 58,
	"endLineNumber": 153,
	"endColumn": 58
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "cpp",
	"severity": 8,
	"message": "'DISPLAY_MIRROR_Y' undeclared (first use in this function); did you mean 'LCD_MIRROR_Y'?",
	"startLineNumber": 153,
	"startColumn": 76,
	"endLineNumber": 153,
	"endColumn": 76
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_SWAP_XY\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 156,
	"startColumn": 7,
	"endLineNumber": 156,
	"endColumn": 22
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_MIRROR_X\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 157,
	"startColumn": 7,
	"endLineNumber": 157,
	"endColumn": 22
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_SWAP_XY\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 160,
	"startColumn": 7,
	"endLineNumber": 160,
	"endColumn": 22
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_MIRROR_X\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 161,
	"startColumn": 7,
	"endLineNumber": 161,
	"endColumn": 22
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_SWAP_XY\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 164,
	"startColumn": 7,
	"endLineNumber": 164,
	"endColumn": 22
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_MIRROR_X\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 165,
	"startColumn": 7,
	"endLineNumber": 165,
	"endColumn": 22
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "cpp",
	"severity": 8,
	"message": "'DISPLAY_WIDTH' undeclared (first use in this function); did you mean 'LCD_WIDTH'?",
	"startLineNumber": 210,
	"startColumn": 22,
	"endLineNumber": 210,
	"endColumn": 22
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_WIDTH\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 210,
	"startColumn": 22,
	"endLineNumber": 210,
	"endColumn": 35
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "cpp",
	"severity": 8,
	"message": "'DISPLAY_HEIGHT' undeclared (first use in this function); did you mean 'LCD_HEIGHT'?",
	"startLineNumber": 211,
	"startColumn": 22,
	"endLineNumber": 211,
	"endColumn": 22
},{
	"resource": "/C:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "C/C++: IntelliSense",
	"code": "20",
	"severity": 8,
	"message": "Der Bezeichner \"\"DISPLAY_HEIGHT\"\" ist nicht definiert.",
	"source": "C/C++",
	"startLineNumber": 211,
	"startColumn": 22,
	"endLineNumber": 211,
	"endColumn": 36
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/esp32_smartdisplay.c",
	"owner": "cpp",
	"severity": 4,
	"message": "implicit declaration of function 'lvgl_lcd_init'; did you mean 'lvgl_touch_init'? [-Wimplicit-function-declaration]",
	"startLineNumber": 219,
	"startColumn": 3,
	"endLineNumber": 219,
	"endColumn": 3
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/lvgl_touch_cst816s_i2c.c",
	"owner": "cpp",
	"severity": 8,
	"message": "'CST816S_IO_I2C_CONFIG_DEV_ADDRESS' undeclared (first use in this function); did you mean 'CST816S_IO_I2C_CONFIG_DEV_ADDRESS_15'?",
	"startLineNumber": 50,
	"startColumn": 21,
	"endLineNumber": 50,
	"endColumn": 21
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/lvgl_touch_cst816s_i2c.c",
	"owner": "cpp",
	"severity": 8,
	"message": "'TOUCH_MIRROR_X' undeclared (first use in this function); did you mean 'LCD_MIRROR_X'?",
	"startLineNumber": 72,
	"startColumn": 57,
	"endLineNumber": 72,
	"endColumn": 57
},{
	"resource": "/c:/Users/gerds/Downloads/esp32-smartdisplay-demo-main/esp32-smartdisplay-demo-main/.pio/libdeps/esp32-2432s022c/esp32_smartdisplay/src/lvgl_touch_cst816s_i2c.c",
	"owner": "cpp",
	"severity": 8,
	"message": "'TOUCH_MIRROR_Y' undeclared (first use in this function); did you mean 'LCD_MIRROR_Y'?",
	"startLineNumber": 72,
	"startColumn": 85,
	"endLineNumber": 72,
	"endColumn": 85
}]

Any clue whats going wrong?

**Comment by rzeldent** [comment]
HI Gerd,

I suspect you're missing the board definitions. Please clone the project with --recurse submodules. This will create a boards directory with boards

**Comment by GerdS1960** [comment]
No, the board-definitions are in place and actual Version from last week.

 

Von: Rene ***@***.***> 
Gesendet: Freitag, 12. April 2024 20:11
An: rzeldent/esp32-smartdisplay ***@***.***>
Cc: GerdS1960 ***@***.***>; Author ***@***.***>
Betreff: Re: [rzeldent/esp32-smartdisplay] ESP32-2432S022C Compile Errors (Discussion #157)

 

HI Gerd,

I suspect you're missing the board definitions. Please clone the project with --recurse submodules. This will create a boards directory with boards

**Comment by rzeldent** [comment]
Hi Gerd,

Have you updated to the latest version of platformIO and the ESP32 SDK?
